app-doc/heirloom-doctools/heirloom-doctools-160308.ebuild

```diff
@@ -0,0 +1,215 @@
+
+  sed -i -e 's:e.h:e.h y.tab.h:' eqn/neqn.d/Makefile.mk
+	sed -i -e 's:y.tab.h y.tab.h:y.tab.h:' eqn/neqn.d/Makefile.mk
+
+	# Move some stuff around to better suit our filesystem layout
+  sed -i -e '/INSTALL.*grap\.defines/s:$(LIBDIR):/usr/share/heirloom/doctools:' \
+		grap/Makefile.mk || die
+	sed -i -e '/GRAPDEFINES/s:LIBDIR:"/usr/share/heirloom/doctools":' \
+		grap/main.c || die
+
+	sed -i -e '/INSTALL.*eign/s:$(LIBDIR):/usr/share/heirloom/doctools:' \
+		ptx/Makefile.mk || die
+	sed -i -e '/\/eign/s:\(LIB\|REF\)DIR:"/usr/share/heirloom/doctools":' \
+		ptx/ptx.c refer/mkey3.c || die
+
+	sed -i -e 's:$(LIBDIR)/vgrindefs:/usr/share/heirloom/doctools/vgrindefs:' \
+		vgrind/Makefile.mk || die
+	sed -i -e '/\/vgrindefs/s:LIBDIR:"/usr/share/heirloom/doctools":' \
+		vgrind/vfontedpr.c || die
+
+	sed -i -e 's:$(REFDIR)/papers:/usr/share/heirloom/reftools/papers:g' \
+		refer/Makefile.mk || die
+	sed -i -e '/\/papers\/Ind/s:REFDIR:"/usr/share/heirloom/reftools":' \
+		refer/refer1.c || die
+
+	# Correct paths for the installed man pages, just to be clean
+	find . \( -name '*.1' -or -name '*.1b' -or -name '*.7' -or -name '*.7b' \) -exec \
+		sed -i \
+			-e "s:/usr/ucblib/grap.defines:/usr/share/heirloom/doctools/grap.defines:" \
+			-e "s:/usr/ucblib/vgrindefs:/usr/share/heirloom/doctools/vgrindefs:" \
+			-e "s:/usr/ucblib/vfontedpr:/usr/libexec/heirloom/doctools/vfontedpr:" \
+			-e "s:/usr/ucblib/tmac/vgrind:/usr/share/heirloom/tmac/vgrind:" \
+			-e "s:/usr/ucblib/eign:/usr/share/heirloom/doctools/eign:" \
+			-e "s:/usr/ucb/:/usr/bin/:g" \
+			-e "s:/usr/ucblib/doctools:/usr/share/heirloom/doctools:g" \
+			-e "s:/usr/ucblib/reftools/papers:/usr/share/heirloom/reftools/papers:" \
+			-e "s:/usr/ucblib/reftools:/usr/libexec/heirloom/reftools:g" \
+		{} +
+
+	eapply_user
+}
+
+src_configure() {
+	append-cppflags -D_GNU_SOURCE
+
+	sed \
+		-e "s:@CFLAGS@:${CFLAGS}:" \
+		-e "s:@CXXFLAGS@:${CXXFLAGS}:" \
+		-e "s:@CPPFLAGS@:${CPPFLAGS}:" \
+		-e "s:@LDFLAGS@:${LDFLAGS}:" \
+		-e "s:@CC@:$(tc-getCC):" \
+		-e "s:@CXX@:$(tc-getCXX):" \
+		-e "s:@RANLIB@:$(tc-getRANLIB):" \
+		-e "s:@libdir@:$(get_libdir):" \
+		"${FILESDIR}"/${PV}.config \
+		> "${S}"/mk.config
+}
+
+src_compile() {
+	emake $(use cxx && echo MPM=mpm) \
+  BINDIR="/usr/bin" \
+	LIBDIR="/usr/share/heirloom/doctools" \
+  PUBDIR="/usr/share/heirloom/doctools/pub" \
+	MANDIR="/usr/share/man" \
+	MACDIR="/usr/share/heirloom/doctools/tmac" \
+	FNTDIR="/usr/share/heirloom/doctools/font" \
+	PSTDIR="/usr/share/heirloom/doctools/font/devpost/postscript" \
+	TABDIR="/usr/share/heirloom/doctools/nterm" \
+	HYPDIR="/usr/share/heirloom/doctools/hyphen" \
+	REFDIR="/usr/share/heirloom/reftools" \
+	|| die
+}
+
+src_install() {
+	dodir /usr/share/heirloom/{doc,ref}tools
+
+	# The build system uses the ROOT variable in place of DESTIDR.
+	emake $(use cxx && echo MPM=mpm) INSTALL="/usr/bin/install" STRIP="/usr/bin/strip" \
+  BINDIR="/usr/bin" \
+	LIBDIR="/usr/share/heirloom/doctools" \
```

Comment:
This big bunch bunch of repeating make options really asks for moving to a separate function.

Also, did you run it with FEATURES=test?